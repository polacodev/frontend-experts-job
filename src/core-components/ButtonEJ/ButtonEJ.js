import React from 'react';
import { Container, Content, Button, Text } from 'native-base';

import styles from './ButtonEJ.style';

const ButtonEJ = ({ title = '', onPress = () => ({}) }) => {
  return (
    <Container style={styles.container}>
      <Content>
        <Button style={styles.button} onPress={onPress}>
          <Text>{title}</Text>
        </Button>
      </Content>
    </Container>
  );
};

export default ButtonEJ;
