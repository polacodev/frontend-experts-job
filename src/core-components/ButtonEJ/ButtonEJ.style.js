import { StyleSheet } from 'react-native';

import color from '../../config/color/color';

const styles = (disable) =>
  StyleSheet.create({
    container: {
      marginTop: 10,
      alignItems: 'center',
    },
    button: {
      width: '100%',
      borderRadius: 5,
      justifyContent: 'center',
      backgroundColor: disable ? color.gray : color.primary,
    },
  });

export default styles;
