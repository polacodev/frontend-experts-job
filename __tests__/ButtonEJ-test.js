import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native';

import ButtonEJ from '../src/core-components/ButtonEJ/ButtonEJ';
jest.useFakeTimers();
describe('<ButtonEJ />', () => {
  test('renders <ButtonEJ /> correctly', () => {
    const tree = renderer
      .create(
        <ButtonEJ disable={true} title="Test Button" onPress={() => ({})} />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should return undefined due the empty props', async () => {
    const { container } = render(<ButtonEJ />);
    expect(container.firstChild).toBeUndefined();
  });
});
