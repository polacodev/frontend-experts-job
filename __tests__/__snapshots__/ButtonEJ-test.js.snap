// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonEJ /> renders <ButtonEJ /> correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "flex": 1,
        "height": 1334,
      },
      Object {
        "alignItems": "center",
        "marginTop": 10,
      },
    ]
  }
>
  <RCTSafeAreaView
    emulateUnlessSupported={true}
    style={
      Object {
        "backgroundColor": "transparent",
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      automaticallyAdjustContentInsets={false}
      contentContainerStyle={
        Array [
          Object {
            "padding": undefined,
          },
          undefined,
        ]
      }
      contentInset={
        Object {
          "bottom": 0,
        }
      }
      enableAutomaticScroll={true}
      enableOnAndroid={false}
      enableResetScrollToCoords={true}
      extraHeight={75}
      extraScrollHeight={0}
      getScrollResponder={[Function]}
      handleOnScroll={[Function]}
      keyboardDismissMode="interactive"
      keyboardOpeningTime={250}
      keyboardShouldPersistTaps="handled"
      keyboardSpace={0}
      onScroll={[Function]}
      resetKeyboardSpace={[Function]}
      resetScrollToCoords={
        Object {
          "x": 0,
          "y": 0,
        }
      }
      scrollEventThrottle={1}
      scrollForExtraHeightOnAndroid={[Function]}
      scrollIntoView={[Function]}
      scrollToEnd={[Function]}
      scrollToFocusedInput={[Function]}
      scrollToPosition={[Function]}
      showsVerticalScrollIndicator={true}
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 1,
        }
      }
      update={[Function]}
      viewIsInsideTabBar={false}
    >
      <View>
        <View
          accessible={true}
          focusable={true}
          forwardedRef={[Function]}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "flex-start",
              "backgroundColor": "#bdbdbd",
              "borderBottomWidth": null,
              "borderColor": "#007aff",
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 2,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 6,
              "paddingTop": 6,
              "shadowColor": undefined,
              "shadowOffset": undefined,
              "shadowOpacity": undefined,
              "shadowRadius": undefined,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#fff",
                "fontFamily": "System",
                "fontSize": 16.5,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 16,
                "paddingRight": 16,
              }
            }
            uppercase={false}
          >
            Test Button
          </Text>
        </View>
      </View>
    </RCTScrollView>
  </RCTSafeAreaView>
</View>
`;
